### ISSUE TITLE
Support the AMQP which is based on Rabbitmq in Ali Cloud

### ISSUE THREAD

**Issue by WillPhoenixLiu** [dotnetcore/CAP#350]
When I connected to the AMQP queue, I need to setup the following 2 options. So I add them in the RabbitMQOptions.

The 2 options:

```
public IList<AuthMechanismFactory> AuthMechanisms { get; set; }
public bool TopologyRecoveryEnabled { get; set; }
```
I was trying to create a PR, but I got request failed with status code: 403.

**Comment by yang-xiaodong** [comment]
Hello, we have improved RabbitMQOptions to provide extensions option to configure the client original configuration, you can using like this:

```
x.UseRabbitMQ(o =>
{
    o.HostName = "localhost";
    o.ConnectionFactoryOptions = opt => { 
        //rabbitmq client ConnectionFactory config
    };
 });
```

**Comment by yang-xiaodong** [comment]
Fixed in v2.5.1
